File: Arrays/third-maximum-number/third_maximum_number.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> third_max(vector<int> &nums)
{
    sort(nums.begin(), nums.end(),greater<int>());
    nums.erase(unique(nums.begin(), nums.end()), nums.end());

    if (nums.size() >= 3)
        return nums[2];
    else
        return nums[0];
}

int main()
{
    return 0;
}

File: Arrays/third-maximum-number/third_maximum_number.py
#  Project : SDE Placement Questions
#  Filename : third_maximum_number.py
#  Author : thameem
#  Current modification time : Mon, 24 Oct 2022 at 8:28 pm India Standard Time
#  Last modified time : Mon, 24 Oct 2022 at 8:28 pm India Standard Time

def third_max(nums: list[int]) -> int:
    nums = list(set(nums))
    nums.sort(reverse=True)

    if len(nums) >= 3:
        return nums[2]
    else:
        return nums[0]


if __name__ == '__main__':
    print(third_max([3, 2, 1]))
    print(third_max([2, 1]))
